presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreHttpRequestDetails.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.thrift;
+
+import com.facebook.airlift.json.ObjectMapperProvider;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.thrift.TException;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class ThriftMetastoreHttpRequestDetails
+{
+    private final String url;
+    private final Map<String, String> headers;
+
+    @JsonCreator
+    public ThriftMetastoreHttpRequestDetails(@JsonProperty("url") String url,
+                                             @JsonProperty("headers") Map<String, String> headers)
```

Comment:
This will fail style check. Make each param to its own line

Reply:
The idea of having a map for headers is that depending on what an org's http endpoint signature and payload is, it could fit in within the parameter specified. Hence. the url and the map of headers. At uber, we have 2 mandatory headers for the http request - 1. the identity of the hms service 2. the identity of the caller client

Reply:
I mean move the first param to its own line